### PR TITLE
dumb: Package dumbplay & split libaldmb, take ownership

### PR DESCRIPTION
**Pull request by ScrelliCopter**
#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

#### Description
This is a second go at https://github.com/void-linux/void-packages/pull/18472 with (in my opinion) a better approach using subpackages instead of build time options.

libaldmb is a separate library that isn't used by the rest of the package and splitting it avoids a bunch of unnecessary X11 & other desktop dependencies on `dumb`.

For the reference player I created a `dumbplay` subpackage which keeps the SDL2 dependency out of the main library package, ~~the dumbout util technically doesn't need it but I figured it's not important enough to create yet another subpackage for.~~ the tiny dumbout util has minimal dependencies and thus probably belongs in the main package.

I also updated the homepage which still pointed to the old pre-fork page.

Checked and couldn't see any other packages using the Allegro 4 integration (or libdumb at all for that matter) so I'm certain this change shouldn't break any existing package.

### CODE REVIEW

**Comment by ScrelliCopter** [comment]
Edited comment, didn't realise there was a new PR format.

**Review by ericonr — COMMENTED** [review]
Is this package really something you'd use in an environment where allegro4's dependencies aren't already installed? Maybe push all the extras (aldumb and dumbplay) into a single subpackage with all extra deps?

I'm not sure this will receive any updates any more, so it's not going to make maintenance much harder, I'd just prefer to be sure it's not adding to many unnecessary subpackages.

Also, please include the rationale you exposed in the PR message in the commit message itself.

File: srcpkgs/dumb/template [hunk 1]
```diff
@@ -31,8 +31,33 @@ dumb-devel_package() {
...
+}
+
+aldumb-devel_package() {
+	depends="aldumb>=${version}_${revision} ${sourcepkg}-devel>=${version}_${revision}"
+	short_desc+=", Allegro4 integration - development files"
+	pkg_install() {
+		vmove usr/lib/libaldmb.so
+		vmove usr/include/aldumb.h
+	}
+}
```
[the remark this answers is not in the document]
Comment (ScrelliCopter) [reply srcpkgs/dumb/template:55]: Makes sense, especially if the only people who are going to be using the Allegro stuff would be building against it.

[the remark this answers is not in the document]
Comment (ScrelliCopter) [reply srcpkgs/dumb/template:55]: I figured giving aldmb a devel package would help make it more orthogonal w/ dumb-devel but figures aldmb is already such a specific use case that eliminating the superfluous subpackage is a better idea, I'll do that.

**Comment by ScrelliCopter** [comment]
> Is this package really something you'd use in an environment where allegro4's dependencies aren't already installed? Maybe push all the extras (aldumb and dumbplay) into a single subpackage with all extra deps?
> 
> I'm not sure this will receive any updates any more, so it's not going to make maintenance much harder, I'd just prefer to be sure it's not adding to many unnecessary subpackages.

Allegro 4 is ancient and none of the packages in Void's repos that depend on `allegro4` use `dumb` nor its optional Allegro 4 integration. Newer versions of Allegro use DUMB directly and don't rely on this old support library.
I only know of one well-known piece of software; Adventure Game Studio; that uses both Allegro 4 & DUMB, but its DUMB appears to be an ancient 0.9.2 version that is built in-tree rather than being pulled from the distro.

I'll concede that the proposed split is more than a little philosophical rather than purely technical, but I can see an (admittedly extremely rare) case for running dumb and dumbout on a headless server where the chains of X11 & desktop sound server dependencies allegro4 would pull in only serve to add unnecessary bloat. IMO that justifies splitting a seldom used glue library.

> 
> Also, please include the rationale you exposed in the PR message in the commit message itself.

Will do on next push.

File: srcpkgs/dumb/template [hunk 2]
```diff
@@ -31,8 +31,33 @@ dumb-devel_package() {
...
 	}
 }
+
+aldumb_package() {
+	depends="${sourcepkg}>=${version}_${revision}"
+	short_desc+=", Allegro4 integration"
+	pkg_install() {
+		vmove "usr/lib/libaldmb.so.*"
+	}
+}
```
[the remark this answers is not in the document]
Comment (ScrelliCopter) [reply srcpkgs/dumb/template:46]: Fair point, may affect users who build forks of old games like https://github.com/carstene1ns/alex4
I'm guessing there's no way to deal with a split on existing installations other than hoping there are no users who actively use aldmb.so from `dumb`? (I severely doubt there are, but still)

**Pull request merged**